Neos.ContentRepository.Core/Classes/ContentGraphFinder.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core;
+
+use Neos\ContentRepository\Core\Projection\ContentGraph\ContentGraphInterface;
+use Neos\ContentRepository\Core\SharedModel\Exception\ContentStreamDoesNotExistYet;
+use Neos\ContentRepository\Core\SharedModel\Exception\WorkspaceDoesNotExist;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+
+/**
+ * A finder for ContentGraphInterface bound to contentStream/Workspace
+ *
+ * @see ContentGraphInterface
+ * @api only for read access during write operations and in services
+ */
+final class ContentGraphFinder implements ContentGraphFinderInterface
+{
+    /**
+     * @var array<string, ContentGraphInterface>
+     */
+    private array $contenGraphInstances = [];
+
+    public function __construct(
+        private readonly ContentGraphFactoryInterface $contentGraphFactory
+    ) {
+    }
+
+    /**
+     * @throws WorkspaceDoesNotExist if there is no workspace with the provided name
+     * @throws ContentStreamDoesNotExistYet if the provided workspace does not resolve to an existing content stream
+     */
+    public function fromWorkspaceName(WorkspaceName $workspaceName): ContentGraphInterface
```

Comment:
i think `for*` or `provideForWorkspaceName` (like the naming from the `WorkspaceProvider` makes more sense?

Reply:
riiiight yeah I actually wanted to do that, if we treat this as a finder it should even be `get` ? unless we would add a check if the workspace exist and return null? Might be an idea and then do `findBy`